src/main/java/org/sharedid/endpoint/config/RouteConfiguration.java

```diff
@@ -0,0 +1,123 @@
+    private String optInTemplate;
+
+    @Bean
+    public HandlerRegistry handlerRegistry(List<Handler<RoutingContext>> handlers) {
+        return new HandlerRegistry(handlers);
+    }
+
+    @Bean
+    public TemplateEngine templateEngine(Vertx vertx) {
+        return HandlebarsTemplateEngine.create(vertx);
+    }
+
+    @Bean
+    public TemplateHandler templateHandler(TemplateEngine templateEngine,
+                                           @Value("${handlers.template.template-dir}") String templateDirectory) {
+        return TemplateHandler.create(templateEngine, templateDirectory, "text/html");
+    }
+
+    @Bean
+    public Router router(Vertx vertx, HandlerRegistry handlerRegistry, TemplateEngine templateEngine) {
+        Router router = Router.router(vertx);
+
+        router.route("/*")
+                .handler(handlerRegistry.get(AddRequestMetricsHandler.class))
+                .handler(
+                        CorsHandler.create(".*.")
+                                .allowCredentials(true)
+                                .allowedHeader("Access-Control-Allow-Method")
+                                .allowedHeader("Access-Control-Allow-Origin")
+                                .allowedHeader("Access-Control-Allow-Credentials")
+                                .allowedHeader("Content-type")
+                                .allowedMethod(HttpMethod.GET)
+                                .allowedMethod(HttpMethod.POST)
+                                .allowedMethod(HttpMethod.OPTIONS));
+
+        router.get("/usync")
+                .handler(handlerRegistry.get(AddDefaultHeadersHandler.class))
+                .handler(handlerRegistry.get(ReadParametersHandler.class))
+                .handler(handlerRegistry.get(ReadSharedIdHandler.class))
+                .handler(handlerRegistry.get(ReadOptOutHandler.class))
+                .handler(handlerRegistry.get(CheckOptOutHandler.class))
+                .handler(handlerRegistry.get(ParseGdprConsentStringHandler.class))
+                .handler(handlerRegistry.get(ReadAuditCookieHandler.class))
+                .handler(handlerRegistry.get(CheckGdprConsentHandler.class))
+                .handler(handlerRegistry.get(CheckCcpaConsentHandler.class))
+                .handler(handlerRegistry.get(CheckVendorHandler.class))
+                .handler(handlerRegistry.get(SetAuditCookieHandler.class))
+                .handler(handlerRegistry.get(SetSharedIdCookieHandler.class))
+                .handler(handlerRegistry.get(RedirectHandler.class))
+                .handler(handlerRegistry.get(RespondDefaultHandler.class));
+
+        router.get("/id")
+                .handler(handlerRegistry.get(AddDefaultHeadersHandler.class))
+                .handler(handlerRegistry.get(ReadParametersHandler.class))
+                .handler(handlerRegistry.get(ReadSharedIdHandler.class))
+                .handler(handlerRegistry.get(CheckGdprConsentHandler.class))
```

Comment:
It looks to me `CheckGdprConsentHandler` needs values been set inside `ParseGdprConsentStringHandler` and `ReadAuditCookieHandler`, if the code doesn't call `ParseGdprConsentStringHandler` and `ReadAuditCookieHandler` inside "/id" route, it can't handle Gdpr check properly.

Reply:
That's true. We've discussed this on our team for internal components. It's a trade-off between composability and data locality. We've found that having handler's that do only one thing are easier to test and string together rather than having a single monolithic handler that takes care of everything. The downside is that there are certain data dependencies that occur when between handlers. In practice with good integration tests these are easy to catch.

Reply:
@coryhammon1 I wasn't clear in my comment. 
I do like the idea that `ParseGdprConsentStringHandler` being separated from `ParseGdprConsentStringHandler` and `ReadAuditCookieHandler`. 
My concern is: for `router.get("/id")` and `router.post("/id")`, it calls `ParseGdprConsentStringHandler` without `ParseGdprConsentStringHandler` and `ReadAuditCookieHandler`. It looks to me `ParseGdprConsentStringHandler` for the `"/id"` doesn't function properly.

Reply:
Misunderstood your comment. Added those handlers to both routes.

Reply:
cool.